scripts/license-checker.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+const { join } = require('path')
+const licenseChecker = require('license-checker')
+
+const ROOT_REPO = join(__dirname, '../')
+/**
+ * For full exhaustive list, We can check the valid licenses list from Kibana
+ * https://github.com/elastic/kibana/blob/dc956a0a79a34a46b353f2a0bef5b18be353220f/src/dev/license_checker/config.js#L22
+ */
+const LICENSE_WHITELIST = [
+  'MIT',
+  'MIT*',
+  'Apache-2.0',
+  'Apache License, Version 2.0',
+  'Public Domain',
+  'ISC',
+  'BSD',
+  'BSD*',
+  'BSD-2-Clause',
+  'BSD-3-Clause',
+  'BSD-3-Clause OR MIT',
+  'AFLv2.1',
+  '(MIT OR Apache-2.0)',
+  '(WTFPL OR MIT)',
+  '(MIT AND Zlib)',
+  '(BSD-2-Clause OR MIT OR Apache-2.0)',
+  '(MIT AND BSD-3-Clause)',
+  'CC0-1.0',
+  'CC-BY-3.0',
+  'CC-BY-4.0',
+  'Unlicense'
+]
+
+/**
+ * Packages that must be ignored during license check
+ */
+const IGNORE_LIST = ['colors@0.6.2']
```

Comment:
Why is this necessary?

Reply:
They have a custom license which is already fixed in colors@1.0 so this is a workaround to avoid the check for older versions.

Reply:
```
license: Custom: http://i.imgur.com/goJdO.png
```